libcamera_app.hpp

```diff
@@ -719,6 +722,25 @@ class LibcameraApp
 			preview_->Show(fd, size, w, h, stride);
 		}
 	}
+	void configureDenoise(bool video_mode)
+	{
+		if (options.denoise == "auto") {
+			controls_.set(controls::draft::NoiseReductionMode,
+				      video_mode ? libcamera::controls::draft::NoiseReductionModeFast
+						 : libcamera::controls::draft::NoiseReductionModeHighQuality);
+		} else {
+			static std::map<std::string, int> denoise_table = {
+				{ "off", libcamera::controls::draft::NoiseReductionModeOff },
+				{ "cdn_off", libcamera::controls::draft::NoiseReductionModeMinimal },
+				{ "cdn_fast", libcamera::controls::draft::NoiseReductionModeFast },
+				{ "cdn_hq", libcamera::controls::draft::NoiseReductionModeHighQuality } };
+
+				if (denoise_table.count(options.denoise))
+					controls_.set(controls::draft::NoiseReductionMode, denoise_table[options.denoise]);
+				else
+					throw std::runtime_error("Invalid denoise mode " + options.denoise);
```

Comment:
It's a matter of taste surely, but I would write this as
```
if (!denoise_table.count(options.denoise))
    throw std::runtime_error("Invalid denoise mode " + options.denoise);

controls_.set(controls::draft::NoiseReductionMode, denoise_table[options.denoise]);
```

That way the normal control flow / the 'good' case continues without extra indentation